tests/common/docker_image.py

```diff
@@ -337,6 +337,53 @@ def wait(self, max_attempts: int = 15, sleep_rate: float = 1.0) -> bool:
         return False
 
 
+class SOEFDockerImage(DockerImage):
+    """Wrapper to SOEF Docker image."""
+
+    def __init__(
+        self, client: DockerClient, addr: str, port: int = 9002,
+    ):
+        """
+        Initialize the SOEF Docker image.
+
+        :param client: the Docker client.
+        :param addr: the address.
+        :param port: the port.
+        """
+        super().__init__(client)
+        self._addr = addr
+        self._port = port
+
+    @property
+    def tag(self) -> str:
```

Comment:
One question is, where do we want to have the image? Upload to DockerHub and pull from there or have it built on every call?

Reply:
The build would happen only once, and the next builds would use the cached Docker image layers, so it takes much less time than the first build.

The approach that pulls the image from DockerHub is easier to implement (basically the current code would already work), so if it is possible to publish the Docker image under `fetchai/` would be better in my opinion.

Reply:
The approach that builds the image manually would have to:

- clone the repository https://github.com/fetchai/soef in a temporary directory
- enter in the temporary directory, and call `docker build`

To save time, the clone should only happen if the image is not available. There should be some way to use the Python package `docker`'s API to check that.